salalib/nagent.cpp

```diff
@@ -1010,9 +1010,11 @@ Point2f Agent::onOcclusionLook(bool wholeisovist, int looktype)
          if (fardist != -1.0) {
             bool cont = true;
             if (looktype == AgentProgram::SEL_OCC_MEMORY) {
-               m_occ_memory.a().add(nigpix);
+               if(std::find(m_occ_memory.a().begin(), m_occ_memory.a().end(), nigpix) == m_occ_memory.a().end()) {

```

Comment:
I was wondering whether we need a sorted vector that also can do uniqueness checks - basically like a set but optimised for in order adding and random access...

Reply:
Maybe just methods that add an element in place in a sorted container, and generally handle this.